vividus-engine/src/main/java/org/vividus/variable/DynamicVariable.java

```diff
@@ -16,8 +16,11 @@
 
 package org.vividus.variable;
 
-@FunctionalInterface
+import java.util.Optional;
+
 public interface DynamicVariable
 {
-    String getValue();
+    Optional<String> calculateValue();
+
+    String getMissingValueErrorMessage();
```

Comment:
I think this approach may limit the implementation when during variable resolution more than one error may appear, maybe we could return some pojo like
```java
    class DynamicVaruableValue 
    {
        private final Optional<Object> value;
        private final String failDetails;
    }

Reply:
do we have examples where the first error don't stop variable resolution process? so several errors can stack up

Reply:
we don't

Reply:
It not necessarily to stack them up. What if the resolution could resolve into different errors? How to process it with the current implementation. I'll look closely, but clipboard text resolution may fail by different reasons